crates/vise-macros/src/metrics.rs

```diff
@@ -261,6 +263,15 @@ impl MetricsImpl {
         })
     }
 
+    fn ensure_no_generics(generics: &Generics) -> syn::Result<()> {
+        if generics.params.is_empty() {
+            Ok(())
+        } else {
+            let message = "Generics are not supported for `derive(Metrics)` macro";
+            Err(syn::Error::new_spanned(generics, message))
+        }
+    }
```

Comment:
Could you maybe make `ensure_no_generics` from `labels.rs` a free function and reuse it?